wasm/validation/context.py

```diff
@@ -0,0 +1,148 @@
+from typing import (
+    Optional,
+    Tuple,
+)
+
+from wasm.datatypes import (
+    FuncType,
+    GlobalType,
+    MemoryType,
+    TableType,
+    ValType,
+)
+from wasm.exceptions import (
+    InvalidModule,
+)
+
+
+class Context:
+    def __init__(self,
+                 *,
+                 types: Tuple[FuncType, ...],
+                 funcs: Tuple[FuncType, ...],
+                 tables: Tuple[TableType, ...],
+                 mems: Tuple[MemoryType, ...],
+                 globals: Tuple[GlobalType, ...],
+                 locals: Tuple[ValType, ...],
+                 labels: Tuple[ValType, ...],
+                 returns: Optional[Tuple[ValType, ...]],
```

Comment:
Did you forget a default argument value of `None` here?

Reply:
No, this was actually intentional.  I want to require the argument, but it can be `None`.  I'll add a comment indicating this.

Reply:
Okay, wasn't sure because the `prime` method doesn't require the argument.  But now I'm realizing that the `prime` method doesn't require *any* argument.

Reply:
I haven't quite figured out the `Context.prime` API yet.  It's an attempt to remove some/most of the mutation that currently occurs during validation but it's a WIP.